### DISCUSSION TITLE
error: could not compile `khronos-egl`, `ash`

### DISCUSSION THREAD

**Discussion by garybcoder** [gfx-rs/wgpu#4286] [General]
I'm trying to work through the examples in the wgpu repository and I'm having trouble with the compiling failing depending on which branch I use. 

If I follow the link on the wgpu crates.io page I get 'trunk'. All of the examples compile and run just fine.

However, if I download any other branch such as '0.17', '0.17.0', '0.17.1', or '0.17.2' - compiling any example with "cargo run" sets off my virus scanner and fails with this message:

`error: linking with `link.exe` failed: exit code: 1104
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.28.29333\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "D:\\Users\\Gary\\AppData\\Local\\Temp\\rustcOxyyHj\\symbols.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\khronos-egl-44b9cdcfe1e0ae06\\build_script_build-44b9cdcfe1e0ae06.build_script_build.d84bb827-cgu.0.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\khronos-egl-44b9cdcfe1e0ae06\\build_script_build-44b9cdcfe1e0ae06.build_script_build.d84bb827-cgu.1.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\khronos-egl-44b9cdcfe1e0ae06\\build_script_build-44b9cdcfe1e0ae06.build_script_build.d84bb827-cgu.2.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\khronos-egl-44b9cdcfe1e0ae06\\build_script_build-44b9cdcfe1e0ae06.build_script_build.d84bb827-cgu.3.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\khronos-egl-44b9cdcfe1e0ae06\\build_script_build-44b9cdcfe1e0ae06.build_script_build.d84bb827-cgu.4.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\khronos-egl-44b9cdcfe1e0ae06\\build_script_build-44b9cdcfe1e0ae06.56lmhxiny68nstc.rcgu.o" "/LIBPATH:D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\deps" "/LIBPATH:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-fdaa63d6970be740.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-a684236a9da37ec6.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-85c151aafb35bf5c.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd_detect-7dbe75e48c9029c0.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-78044ac8790346aa.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libminiz_oxide-8743285c70baade8.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libadler-b260ead09cc3ca75.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-200564555b6c364f.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-60e4ae6d068e334f.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-d74559c1731cb9bf.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-3d51e2979c44c27c.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-268ac0a196c81649.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-e74895db82cfb265.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-d805a250e2b7e197.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-3f507ba7adac1c2a.rlib" "kernel32.lib" "advapi32.lib" "userenv.lib" "kernel32.lib" "ws2_32.lib" "bcrypt.lib" "msvcrt.lib" "legacy_stdio_definitions.lib" "/NXCOMPAT" "/LIBPATH:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\khronos-egl-44b9cdcfe1e0ae06\\build_script_build-44b9cdcfe1e0ae06.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
  = note: LINK : fatal error LNK1104: cannot open file 'D:\Rust\___rust\crates\_graphics\wgpu\wgpu-0.17.1\wgpu-0.17.1\target\debug\build\khronos-egl-44b9cdcfe1e0ae06\build_script_build-44b9cdcfe1e0ae06.exe'
          

error: could not compile `khronos-egl` due to previous error
warning: build failed, waiting for other jobs to finish...
error: linking with `link.exe` failed: exit code: 1104
  |
  = note: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.28.29333\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "D:\\Users\\Gary\\AppData\\Local\\Temp\\rustcjU7tJ5\\symbols.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\ash-9f2b5ea704b2fda5\\build_script_build-9f2b5ea704b2fda5.build_script_build.dd456809-cgu.0.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\ash-9f2b5ea704b2fda5\\build_script_build-9f2b5ea704b2fda5.build_script_build.dd456809-cgu.1.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\ash-9f2b5ea704b2fda5\\build_script_build-9f2b5ea704b2fda5.build_script_build.dd456809-cgu.2.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\ash-9f2b5ea704b2fda5\\build_script_build-9f2b5ea704b2fda5.build_script_build.dd456809-cgu.3.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\ash-9f2b5ea704b2fda5\\build_script_build-9f2b5ea704b2fda5.build_script_build.dd456809-cgu.4.rcgu.o" "D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\ash-9f2b5ea704b2fda5\\build_script_build-9f2b5ea704b2fda5.5binkutvjwdsj2y9.rcgu.o" "/LIBPATH:D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\deps" "/LIBPATH:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-fdaa63d6970be740.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-a684236a9da37ec6.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-85c151aafb35bf5c.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd_detect-7dbe75e48c9029c0.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-78044ac8790346aa.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libminiz_oxide-8743285c70baade8.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libadler-b260ead09cc3ca75.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-200564555b6c364f.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-60e4ae6d068e334f.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-d74559c1731cb9bf.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-3d51e2979c44c27c.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-268ac0a196c81649.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-e74895db82cfb265.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-d805a250e2b7e197.rlib" "D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-3f507ba7adac1c2a.rlib" "kernel32.lib" "advapi32.lib" "userenv.lib" "kernel32.lib" "ws2_32.lib" "bcrypt.lib" "msvcrt.lib" "legacy_stdio_definitions.lib" "/NXCOMPAT" "/LIBPATH:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:D:\\Rust\\___rust\\crates\\_graphics\\wgpu\\wgpu-0.17.1\\wgpu-0.17.1\\target\\debug\\build\\ash-9f2b5ea704b2fda5\\build_script_build-9f2b5ea704b2fda5.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:D:\\Users\\Gary\\.rustup\\toolchains\\1.65-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
  = note: LINK : fatal error LNK1104: cannot open file 'D:\Rust\___rust\crates\_graphics\wgpu\wgpu-0.17.1\wgpu-0.17.1\target\debug\build\ash-9f2b5ea704b2fda5\build_script_build-9f2b5ea704b2fda5.exe'
          

error: could not compile `ash` due to previous error
>Exit code: 101
`

I have no way of determining whether this is a false position or not. My virus scanner, BitDefender, says that it is infected with Trojan.GenericKDZ.95853.

Why would any 0.17 branch cause this problem, but the trunk does not?

Thank, Gary

**Comment by cwfitzgerald** [comment]
Virus scanners are a crap shoot - maybe try updating the definitions in your virus scanner. Alternatively if you have a folder you use for programming, add that to the exception list.

**Reply by garybcoder** [reply]
When I completely disable the virus scanner the anti-virus no longer complains; but "cargo run" still fails, even if I delete the target folder.
`
error: linking with `link.exe` failed: exit code: 1104
 = note: LINK : fatal error LNK1104: cannot open file 'D:\Rust\___rust\crates\_graphics\wgpu\wgpu-0.17.1\wgpu-0.17.1\target\debug\build\khronos-egl-44b9cdcfe1e0ae06\build_script_build-44b9cdcfe1e0ae06.exe'
error: could not compile `khronos-egl` due to previous error

error: linking with `link.exe` failed: exit code: 1104
  = note: LINK : fatal error LNK1104: cannot open file 'D:\Rust\___rust\crates\_graphics\wgpu\wgpu-0.17.1\wgpu-0.17.1\target\debug\build\ash-9f2b5ea704b2fda5\build_script_build-9f2b5ea704b2fda5.exe'
error: could not compile `ash` due to previous error
`
However, if I delete the entire folder and extract the repository from the zip file that I downloaded - then it compiles and runs just fine.

I'm not sure why I would have to do that, but it is working now.

**Reply by cwfitzgerald** [reply]
Almost certainly the issue was that the exe got something weird done to it by the antivirus, meaning that cargo couldn't overwrite it.
